azure/services/virtualmachines/virtualmachines.go

```diff
@@ -83,6 +83,7 @@ func (s *Service) Reconcile(ctx context.Context) error {
 
 	result, err := async.CreateResource(ctx, s.Scope, s.Client, vmSpec, serviceName)
 	s.Scope.UpdatePutStatus(infrav1.VMRunningCondition, serviceName, err)
+	s.Scope.UpdatePutStatus(infrav1.DisksReadyCondition, serviceName, err)
```

Comment:
I think we should consider only setting the Disk condition once the VM create is successful. The reason I say that is that if the overall operation fails, it's hard to know whether it's caused by the VM or the disk, and if we set the disk ready condition to `false` it could potentially shadow the VM ready condition in the overall Machine status.
Eg scenario: User creates a Machine. Azure VM provisioning fails. The AzureMachine shows `Ready` as "False" and reason as "disk failed to create". This is confusing because the disk reconcile is a no-op if they look at the code. It would be better to always show "vm failed to create" in those cases.

Reply:
let's also add a comment to explain we set the disk condition in the VM service